packages/theme/src/cli/commands/theme/push.ts

```diff
@@ -155,7 +149,7 @@ export default class Push extends ThemeCommand {
       return
     }
 
-    if (flags['dev-preview'] || (!flags.stable && !flags.password)) {
+    if (!flags.stable && !flags.password) {
```

Comment:
I'm keeping the `password` check for now, meaning that if the `password` flag is provided we will still use the Ruby CLI.

This check was added to avoid breaking the `push` command for CI users - more context [here](https://github.com/Shopify/cli/pull/3765)

I think the changes introduced in the linked PR above should address those issues, however since there are numerous changes here and many users 'set and forget' their CI configs, I think it would be prudent to de-risk this PR by leaving this here.

No strong opinions on this though